src/components/CreateNewProjectWizard/MainPane.js

```diff
@@ -29,29 +31,116 @@ type Props = {
   handleSubmit: () => Promise<any> | void,
 };
 
-class MainPane extends PureComponent<Props> {
+type State = {
+  gatsbyStarter: string, // Temporary value during selection in selection toast
+};
+
+class MainPane extends PureComponent<Props, State> {
+  state = {
+    gatsbyStarter: '',
+  };
+
   handleFocusProjectName = () => this.props.focusField('projectName');
   handleBlurProjectName = () => this.props.focusField(null);
+  handleFocusStarter = () => this.props.focusField('projectStarter');
 
   updateProjectName = (projectName: string) =>
     this.props.updateFieldValue('projectName', projectName);
   updateProjectType = (projectType: ProjectType) =>
     this.props.updateFieldValue('projectType', projectType);
   updateProjectIcon = (projectIcon: string) =>
     this.props.updateFieldValue('projectIcon', projectIcon);
+  updateGatsbyStarter = (selectedStarter: string) =>
+    this.props.updateFieldValue('projectStarter', selectedStarter);
 
+  projectSpecificSteps() {
+    const { activeField, projectType, projectStarter } = this.props;
+    switch (projectType) {
+      case 'gatsby':
+        return (
+          <FadeIn key="step-starter">
+            <FormField
+              label="Project Starter"
+              isFocused={activeField === 'projectStarter'}
+            >
+              <ProjectStarterSelection
+                isFocused={activeField === 'projectStarter'}
+                handleFocus={this.handleFocusStarter}
+                onSelect={this.updateGatsbyStarter}
+                projectStarter={projectStarter}
+              />
+            </FormField>
+          </FadeIn>
+        );
+      default:
+        return null;
+    }
+  }
+
+  renderConditionalSteps(currentStepIndex: number) {
+    const { activeField, projectType, projectIcon } = this.props;
+    const buildSteps: Array<?React$Node> = [
+      // currentStepIndex = 0
+      <FadeIn key="step-type">
+        <FormField
+          label="Project Type"
+          isFocused={activeField === 'projectType'}
+        >
+          <ProjectTypeSelection
+            projectType={projectType}
+            onProjectTypeSelect={selectedProjectType =>
+              this.updateProjectType(selectedProjectType)
+            }
+          />
+        </FormField>
+      </FadeIn>,
+      this.projectSpecificSteps(), // currentStepIndex = 1
+      <FadeIn key="step-icon">
+        <FormField
+          label="Project Icon"
+          focusOnClick={false}
+          isFocused={activeField === 'projectIcon'}
+        >
+          <ProjectIconSelection
+            selectedIcon={projectIcon}
+            randomize={true}
+            limitTo={8}
+            onSelectIcon={this.updateProjectIcon}
+          />
+        </FormField>
+      </FadeIn>,
+    ].filter(step => !!step);
+
+    const renderedSteps: Array<?React$Node> = buildSteps.slice(
+      0,
+      currentStepIndex
+    );
```

Comment:
I feel like this might be a bit over-engineered... I think I'd opt for a "dumber" approach:

```
const steps = [];

if (currentStepIndex > 0) {
  steps.push(/* first step */);
}

if (currentStep > 1 && projectType === 'gatsby') {
  steps.push(/* project-specific step for Gatsby */);
}

if (currentStep > 2) {
  steps.push(/* third step */);
}
```

In addition to being a bit easier to follow, it doesn't create a bunch of elements that might not be used.

Reply:
I think you're right. That's a bit over-engineered. :-) Your code is probably easier to follow. I'll have a look at it and maybe I re-write this but I have to think about how I'll implement your proposed flow with-out the additional modal.
So if I'm changing the step order I have to modify this part.